subsys/debug/tracing/cpu_stats.c

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <tracing_cpu_stats.h>
+#include <misc/printk.h>
+
+enum cpu_state {
+	CPU_STATE_IDLE,
+	CPU_STATE_NON_IDLE,
+	CPU_STATE_SCHEDULER
+};
+
+static enum cpu_state last_cpu_state = CPU_STATE_SCHEDULER;
+static enum cpu_state cpu_state_before_interrupts;
+
+static u32_t last_time;
+static u64_t idle_cnt;
+static u64_t non_idle_cnt;
+static u64_t sched_cnt;
+static int nested_interrupts;
+static struct k_thread *current_thread;
+
+#ifndef CONFIG_SMP
+extern k_tid_t const _idle_thread;
+#endif
+
+static inline int is_idle_thread(struct k_thread *thread)
+{
+#ifdef CONFIG_SMP
+	return thread->base.is_idle;
+#else
+	return thread == _idle_thread;
+#endif
+}
+
+void increase_counter(volatile u64_t *cnt)
```

Comment:
What about update_counter()? Then the naming will match the cpu_stats_update_counters().

Reply:
done